src/cli.c

```diff
@@ -28,153 +28,228 @@
+                exit_code = EXIT_INVALID_ARG;
+                goto exit;
             }
-            if (opt == 'q' || !strcmp(longopt, "quit")) {
-                empty_run++;
-                continue;
+            if (opt.key == 'i' || is_longopt(opt, "interactive")) {
+                flags.interactive = true;
+                break;
             }
-            if (!strcmp(longopt, "strict-module-detection")) {
-                strict_module_detection++;
-                continue;
+            if (opt.key == 'q' || is_longopt(opt, "quit")) {
+                flags.empty_run = true;
+                break;
             }
-            if (opt) {
-                fprintf(stderr, "tjs: unknown option '-%c'\n", opt);
-            } else {
-                fprintf(stderr, "tjs: unknown option '--%s'\n", longopt);
+            if (is_longopt(opt, "strict-module-detection") || is_longopt(opt, "strictModuleDetection")) {
+                flags.strict_module_detection = true;
+                break;
             }
-            help();
+            report_unknown_option(&opt);
+            exit_code = EXIT_INVALID_ARG;
+            goto exit;
         }
     }
 
     qrt = TJS_NewRuntime();
     ctx = TJS_GetJSContext(qrt);
 
-    if (!empty_run) {
-        if (expr) {
-            if (eval_buf(ctx, expr, strlen(expr), "<cmdline>", 0))
-                goto fail;
-        } else if (optind >= argc) {
-            /* interactive mode */
-            interactive = 1;
-        } else {
-            const char *filename;
-            filename = argv[optind];
-            int flags = JS_EVAL_TYPE_MODULE;
-            if (strict_module_detection && !has_suffix(filename, ".mjs")) {
-                flags = JS_EVAL_TYPE_GLOBAL;
-            }
-            JSValue ret = TJS_EvalFile(ctx, filename, flags, true, override_filename);
-            if (JS_IsException(ret)) {
-                tjs_dump_error(ctx);
-                JS_FreeValue(ctx, ret);
-                goto fail;
-            }
-            JS_FreeValue(ctx, ret);
+    if (flags.empty_run)
+        goto exit;
+
+    if (flags.eval_expr) {
+        if (eval_buf(ctx, flags.eval_expr, "<cmdline>", JS_EVAL_TYPE_GLOBAL)) {
+            exit_code = EXIT_FAILURE;
+            goto exit;
         }
-        if (interactive) {
-            TJS_RunRepl(ctx);
+    } else if (optind >= argc) {
+        /* interactive mode */
+        flags.interactive = true;
+    } else {
+        const char *filepath = argv[optind];
+        int eval_flags = JS_EVAL_TYPE_MODULE;
+        if (flags.strict_module_detection && !has_suffix(filepath, ".mjs"))
+            eval_flags = JS_EVAL_TYPE_GLOBAL;
+        if (eval_module(ctx, filepath, flags.override_filename, eval_flags)) {
+            exit_code = EXIT_FAILURE;
+            goto exit;
         }
-        TJS_Run(qrt);
     }
 
-    TJS_FreeRuntime(qrt);
-    return 0;
-fail:
-    TJS_FreeRuntime(qrt);
-    return 1;
+    if (flags.interactive)
```

Comment:
Add braces (I know they are not needed, but conistency!). I tried to force thi with clang-format, but I don't know how :-/

Reply:
Ok, let's learn how to force this using clang.

Reply:
Or not, seems there is no way 🤦‍♂